features/netsocket/TLSSocket.h

```diff
@@ -94,8 +97,74 @@ class TLSSocket : public TLSSocketWrapper {
 private:
     TCPSocket tcp_socket;
 };
-
 #endif // MBEDTLS_SSL_CLI_C
+
+#else // MBED_CONF_NSAPI_OFFLOAD_TLSSOCKET
+
+class TLSSocket : public TCPSocket
+{
+public:
+    TLSSocket();
+    virtual ~TLSSocket();
+
+    /** Set hostname.
+     *
+     * TLSSocket requires hostname used to verify the certificate.
+     * If hostname is not given in constructor, this function must be used before
+     * starting the TLS handshake.
+     *
+     * @param hostname     Hostname of the remote host, used for certificate checking.
+     */
+    nsapi_error_t set_hostname(const char *hostname);
+
+    /** Sets the certification of Root CA.
+     *
+     * @note Must be called after open() before calling connect()
+     *
+     * @param root_ca Root CA Certificate in any Mbed TLS-supported format.
+     * @param len     Length of certificate (including terminating 0 for PEM).
+     * @return        0 on success, negative error code on failure.
```

Comment:
Should we stick to the `NSAPI_ERROR_OK` convention, rather than disclose its value `0`?

Reply:
Good finding. I'll provide a fix.